src/Swarm/Failure/Render.hs

```diff
@@ -0,0 +1,37 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Swarm.Failure.Render where
```

Comment:
@kostmo @byorgey why is this split from the `Failure` module? It depends on nothing else.

AFAIK we do not have any other modules that have data declaration and basic functions separately.

Look at `Exception.hs` - it has `data Exn` _and_ `formatExn` in the same module.

Reply:
I was just commenting on a better place for the module to be, but I agree, it can/should probably just go in the `Failure` module itself.

Reply:
In this case I had intended to use quoting functions from `Swarm.Util` in the rendering code.  And up until 75669d7, `Swarm.Util` depended on types in the `Failure` module, causing a circular dependency.  So I did the work of splitting the modules, but later decided I didn't need the quoting functions after all.

However, after doing that splitting work, I'm not particularly inclined to merge `Swarm.Failure` and `Swarm.Failure.Render` back into a single module again---IMO, there is no downside to the types and the rendering functions being in separate modules, even if there is no technical requirement to be so.  I've found it to be a common step in the evolution of a module: auxiliary functions that had been self-contained with the data definitions eventually needing to pull in an external dependency, requiring the data to move to a separate "types" module to break a circular reference.  It may very well become necessary again for `Failure` rendering at some point.

Reply:
@kostmo that is inconsistent with the code base style though.

Rendering widgets - sure.

But pure functions that are critical to usability of the type (`ParseError` and `prettyParseError`) can absolutely go into the same module and they do everywhere else in Swarm.

Which language puts pure functions on data in different module? In C++ and Python, I have always seen the functions that are basically “methods” in the same module.

Regarding dependencies, your reasoning is backwards.  The Util module should simply not import anything else. It has to be on the bottom of dependency graph. We should try to move stuff _from_ it (as you did) instead of importing modules _in_ it.

Reply:
Finally from a practical standpoint this is premature and silly.

We have modules 2000+loc and argue here for splitting a 100loc module in a novel way, based on OOP best practise.

The future split could go another way though - split failures based on which part of game or UI produced it.
Whoever adds more code to failures in the future can decide which split fits better.

Reply:
As a side note, the smart constructor use case is a good point, but _it does not apply here_.

We will be adding failure types and their pretty printing at the same time, not tweaking the pretty printing function. This is not about "rendering a widget" - I am not going to change the colour of exceptions tomorrow and recompile the whole code base because `Text` does not have a colour `Attr` that I could change.

What I am trying to point out is that failure data type and format are tightly linked, think of it like C++ [`e.what()`](https://en.cppreference.com/w/cpp/error/exception/what) method instead of `failureWidgetWithAttr`.